flo-workflow/src/main/java/com/spotify/flo/context/InstrumentedContext.java

```diff
@@ -125,10 +125,27 @@ default void close() throws IOException {}
   }
 
   private final Listener listener;
+  private TaskOperator.Listener taskOperatorListener;
 
   private InstrumentedContext(EvalContext baseContext, Listener listener) {
     super(baseContext);
     this.listener = Objects.requireNonNull(listener);
+    this.taskOperatorListener = super.listener().composeWith(wrapListener(listener));
+  }
+
+  // this is static to avoid capturing "this" in the anonymous class
```

Comment:
@NatashaL this is the tricky part. we don't want to pass those contexts to subprocess.

Reply:
How about not having this be an anonymous class? Make it a static class instead?